src/Aspire.Dashboard/wwwroot/js/app.js

```diff
@@ -53,15 +53,33 @@ function isScrolledToBottom(container) {
 }
 
 window.copyTextToClipboard = function (id, text, precopy, postcopy) {
-    let tooltipDiv = document.querySelector('fluent-tooltip[anchor="' + id + '"]').children[0];
+    let button = document.querySelector(`#${id}`);
+
+    // If there is a pending timeout then clear it. Otherwise the pending timeout will prematurely reset values.
+    if (button.dataset.copyTimeout) {
+        clearTimeout(button.dataset.copyTimeout);
+        delete button.dataset.copyTimeout;
```

Comment:
`copyTimeout` is just a `number` (specifically an integer), do we need to delete it? setting it to null, undefined or even 0 would probably still work. I'm curious if this is something specific for the result of setTimeout or not.

Reply:
Don't need to use delete. But I did because `button.dataset` is basically a dictionary, and delete is the closest thing to .NET's `dictionary.Remove(key)`.